src/core/public/chrome/ui/header/header_help_menu.tsx

```diff
@@ -81,35 +87,36 @@ export type ChromeHelpExtensionMenuDocumentationLink = EuiButtonEmptyProps & {
    * i.e. `${ELASTIC_WEBSITE_URL}guide/en/kibana/${DOC_LINK_VERSION}/${appName}.html`,
    */
   href: string;
-};
+}
 
 /** @public */
-export type ChromeHelpExtensionMenuCustomLink = EuiButtonEmptyProps & {
+export interface ChromeHelpExtensionMenuCustomLink extends ChromeHelpExtensionLinkBase {
   /**
    * Extend EuiButtonEmpty to provide extra functionality
    */
   linkType: 'custom';
+  /**
+   * URL of the link
+   */
+  href: string;
   /**
    * Content of the button (in lieu of `children`)
    */
   content: React.ReactNode;
-};
+}
 
 /** @public */
-export type ChromeHelpExtensionMenuLink = ExclusiveUnion<
-  ChromeHelpExtensionMenuGitHubLink,
-  ExclusiveUnion<
-    ChromeHelpExtensionMenuDiscussLink,
-    ExclusiveUnion<ChromeHelpExtensionMenuDocumentationLink, ChromeHelpExtensionMenuCustomLink>
-  >
->;
+export type ChromeHelpExtensionMenuLink =
+  | ChromeHelpExtensionMenuGitHubLink
+  | ChromeHelpExtensionMenuDiscussLink
+  | ChromeHelpExtensionMenuDocumentationLink
+  | ChromeHelpExtensionMenuCustomLink;
```

Comment:
Removing the `ExclusiveUnion` adds a (little) more boilerplate in the `switch type` logic in `renderCustomContent`, but it's way more readable, and the tsdoc is greatly improved.